node_modules/oae-zencoder/lib/api.js

```diff
@@ -0,0 +1,137 @@
+ *
+ * @return {Object} Zencoder configuration
+ */
+var init = module.exports.init = function(_config) {
+    config = _config;
+};
+
+/**
+ * @return {Object}    The configuration object that is currently in use.
+ */
+var getConfiguration = module.exports.getConfiguration = function() {
+    return config;
+};
+
+/**
+ * Save data for the completed preview when encoding has been completed.
+ *
+ * @param  {Object}     notification    The notification data received from Zencoder
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ */
+
+var encodingCompleted = module.exports.encodingCompleted = function(notification, callback) {
+    log().info('Received encoding completed notification from Zencoder');
+    var job = notification.job;
+
+    if (job.state === 'finished') {
+        var output = notification.output;
+        var zencoderId = output.id.toString();
+
+        ZencoderDAO.getContentRevisionId(zencoderId, function(err, contentRevision) {
+            if (err) {
+                return callback(err);
+            } else if (!contentRevision || !contentRevision.contentId || !contentRevision.revisionId) {
+                log().warn('No preview job exists for zencoderId %s', zencoderId);
+                return callback('Non-existing Zencoder ID was provided');
+            }
+
+            log().info('Encoding of video content with the contentId %s finished successfully.', contentRevision.contentId);
+
+            _createAndSavePreviews(contentRevision, output, function(err) {
+                if (err) {
+                    return callback(err);
+                }
+
+                return callback();
+            });
+        });
+    }
+
+    else if (job.state === 'failed') {
+        return callback({'err': notification, 'msg': 'Zencoder could not transcode this video for preview'});
+    }
+
+    else {
+        return callback({'err': notification, 'msg': 'Could not process notification from Zencoder'});
+    }
+};
+
+/**
+ * Create and save the previews
+ *
+ * @param  {Object}     contentRevision   An object containing content item and revision IDs
+ * @param  {Object}     output            The output information from Zencoder
+ * @param  {Function}   callback          Standard callback function
+ * @param  {Object}     callback.err      An error that occurred, if any
+ * @api private
+ */
+var _createAndSavePreviews = function(contentRevision, output, callback) {
+    var ctx = new PreviewContext(PreviewProcessorAPI.getConfiguration(), contentRevision.contentId, contentRevision.revisionId);
+
+    ctx.login(function(err) {
+        if (err) {
+            return callback(err);
+        }
+
+        ctx.getContentData(function(err) {
+            if (ctx.revision.previews.total > 0) {
+                return callback('Previews have already been created for this revision');
```

Comment:
This error should be a `{'code': n, 'msg': 'Previews have already...'}` style of error.